### PR TITLE
gh-89610: Add .pyi as a recognised extension for IDLE on macOS

### PR DESCRIPTION
**Pull request by erlend-aasland**

### CODE REVIEW

File: Lib/idlelib/NEWS.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@ What's New in IDLE 3.11.0
 Released on 2022-10-03
 =========================
 
+gh-89610: IDLE on macOS now recognise `.pyi` extensions.
```
Comment (terryjreedy) [inline Lib/idlelib/NEWS.txt]: There are obviously differences of opinion, but I prefer active voice for first sentence of blurbs.
Note: the American spelling uses 'z' versus 's' here and many other places.  The GH spell corrector sees 's' as a misspelling.
```suggestion
gh-89610: Add .pyi as a recognized extension for IDLE on macOS.  This allows
opening stub files by double clicking on them in the Finder.
```

File: Misc/NEWS.d/next/IDLE/2022-07-28-18-56-57.gh-issue-89610.hcosiM.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+IDLE on macOS now recognise ``.pyi`` extensions.
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/IDLE/2022-07-28-18-56-57.gh-issue-89610.hcosiM.rst]:
```suggestion
Add .pyi as a recognized extension for IDLE on macOS.    This allows opening
stub files by double clicking on them in the Finder.
```

**Review by terryjreedy — APPROVED** [review]
Approved after changes.  I will merge after CI runs.

File: Lib/idlelib/NEWS.txt (hunk 1 above)
Comment (erlend-aasland) [reply Lib/idlelib/NEWS.txt]: Noted for future PRs; thanks

**Comment by erlend-aasland** [comment]
Thanks for the reviews, Alex, Éric, and Terry!

**Pull request merged**